Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs

```diff
@@ -642,7 +646,9 @@ private async Task<string> GetToken(ListingsProviderInfo info, CancellationToken
             CancellationToken cancellationToken)
         {
             using var options = new HttpRequestMessage(HttpMethod.Post, ApiUrl + "/token");
-            options.Content = new StringContent("{\"username\":\"" + username + "\",\"password\":\"" + password + "\"}", Encoding.UTF8, MediaTypeNames.Application.Json);
+            var hashedPasswordBytes = _cryptoProvider.ComputeHash("SHA1", Encoding.ASCII.GetBytes(password), Array.Empty<byte>());
+            string hashedPassword = string.Concat(hashedPasswordBytes.Select(b => b.ToString("x2", CultureInfo.InvariantCulture)));
```

Comment:
Cmon I wrote this (imo) beautiful (and fast) [Hex](https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Common/Hex.cs) class :p

Reply:
Switched